c/extras/py_ftrl.cc

```diff
@@ -0,0 +1,413 @@
+    DataTable* dt_target = fm->predict(dt_test).release();
+    py::oobj df_target = py::oobj::from_new_reference(py::Frame::from_datatable(dt_target));
+    return df_target;
+  } else {
+    throw ValueError() << "Cannot make any predictions, because the model was not trained";
+  }
+}
+
+
+PKArgs Ftrl::Type::args_reset(0, 0, 0, false, false, {}, "reset",
+R"(reset(self)
+--
+
+Reset an FTRL model.
+
+Parameters
+----------
+    None
+
+Returns
+----------
+    None
+)");
+
+
+void Ftrl::reset(const PKArgs&) {
+  fm->init_model();
+}
+
+/*
+*  Getter and setter for the model datatable.
+*/
+oobj Ftrl::get_model() const {
+  if (fm->is_trained()) {
+    DataTable* dt_model = fm->get_model();
+    py::oobj df_model = py::oobj::from_new_reference(py::Frame::from_datatable(dt_model));
+    return df_model;
+  } else {
+    return py::None();
+  }
+}
+
+
+void Ftrl::set_model(robj model) {
+  DataTable* dt_model_in = model.to_frame();
+  const std::vector<std::string>& model_cols_in = dt_model_in->get_names();
+
+  if (dt_model_in->nrows != fm->get_d() || dt_model_in->ncols != 2) {
+    throw ValueError() << "FTRL model frame must have " << fm->get_d() << " rows,"
+                       << "and 2 columns, whereas your frame has " << dt_model_in->nrows
+                       << " rows and " << dt_model_in->ncols << " columns";
+  }
+
+  if (model_cols_in != FtrlModel::model_cols) {
+    throw ValueError() << "FTRL model frame must have columns named `z` and `n`,"
+                       << "whereas your frame has the following column names `" << model_cols_in[0]
+                       << "` and `" << model_cols_in[1] << "`";
+  }
+
+  if (dt_model_in->columns[0]->stype() != SType::FLOAT64 ||
+    dt_model_in->columns[1]->stype() != SType::FLOAT64) {
+    throw ValueError() << "FTRL model frame must have both column types as `float64`, "
+                       << "whereas your frame has the following column types: `"
+                       << dt_model_in->columns[0]->stype()
+                       << "` and `" << dt_model_in->columns[1]->stype() << "`";
+  }
+
+  fm->set_model(dt_model_in);
+}
+
+
+/*
+*  All other getters and setters.
+*/
+oobj Ftrl::get_a() const {
+  return py::ofloat(fm->get_a());
+}
+
+
+oobj Ftrl::get_b() const {
+  return py::ofloat(fm->get_b());
+}
+
+
+oobj Ftrl::get_l1() const {
+  return py::ofloat(fm->get_l1());
+}
+
+
+oobj Ftrl::get_l2() const {
+  return py::ofloat(fm->get_l2());
+}
+
+
+oobj Ftrl::get_d() const {
+  return py::oint(static_cast<size_t>(fm->get_d()));
+}
+
+
+oobj Ftrl::get_n_epochs() const {
+  return py::oint(fm->get_n_epochs());
+}
+
+
+oobj Ftrl::get_inter() const {
+  return py::oint(static_cast<size_t>(fm->get_inter()));
+}
+
+
+oobj Ftrl::get_hash_type() const {
+  return py::oint(static_cast<size_t>(fm->get_hash_type()));
+}
+
+
+oobj Ftrl::get_seed() const {
+  return py::oint(static_cast<size_t>(fm->get_seed()));
+}
+
+
+void Ftrl::set_a(robj a) {
+  if (!a.is_numeric()) {
```

Comment:
`a.to_double()` will do the numeric check, no need to do it manually

Reply:
Yes, but this will have a message that will specify which parameter is not numeric, while the `a.to_double()` will only have a some generic error, right?